files/.github/workflows/main.yml

```
@@ -1,5 +1,5 @@
 name: CI
-on: [push]
+on: [push, pull_request]
```

This way, CI will run twice, because it'll be triggered both by the push to the branch and the PR itself.

You can fix it like that though:

```suggestion
on:
  push:
    branches:
      - master
  pull_request:
```

That way, the CI will run on pull request changes and on push to master, but not on push to other branches.

---

Hmm okay, but somehow I'd like to run CI also on branches without opening a PR (yet). At least on software projects I often do experiments on branches without ever opening a PR, or fixing any CI failures *before* opening a PR yet (for example to avoid email spam due to many fixup pushes).

Would it be a problem if CI runs twice? AFAIK Travis-CI and AppVeyor also run twice in that case :winK:

---

Yeah, if that's a requirement then your config is correct. I usually just open a draft PR to get the CI to run.

---

OK so then I'll keep it as-is :+1:

---

I'm still puzzled why most CI systems can't detect that they already ran for a given commit if they were triggered before by a branch push and then again by a pull request.